### ISSUE TITLE
Return value not checked for functions NC4_get_fileinfo() and NC4_put_propattr()

### ISSUE THREAD

**Issue by edhartnett** [Unidata/netcdf-c#747]
In nc4file.c, function nc4_create_file() we have this code (comments in my branch, not yet on master):

```
   /* Get the HDF5 superblock and read and parse the special
    * _NCProperties attribute. */
   NC4_get_fileinfo(nc4_info, &globalpropinfo);

   /* Write the _NCProperties attribute. */
   NC4_put_propattr(nc4_info);

```

At the bottom of nc4_open_file() we have:

```
   /* Get the HDF5 superblock and read and parse the special
    * _NCProperties attribute. */
   NC4_get_fileinfo(nc4_info, NULL);


```

Both NC4_get_fileinfo() and NC4_put_propattr() can return error codes, but we are not checking for that.

When I put in checking, I get a failure. @DennisHeimbigner is it expected that either of these functions will return non-zero in correct operation?

**Comment by DennisHeimbigner** [comment]
the question is what to  do if we cannot wrire this special attribute. ideally we generate a
warning and pcontinue.

**Comment by edhartnett** [comment]
Do we know what conditions we will not be able to create it or read it? That is, under what circumstances we expect them to fail?

I have not studied why or where an error was returned. Just wondering if it is expected. If so, I could document that.

**Comment by DennisHeimbigner** [comment]
good idea. i will do that.

**Comment by edhartnett** [comment]
OK, whatever was causing the failure was cleaned up in some other way. I just put in return code checking for these two calls and tests all pass, which is good and as expected.

I have merged this into open PR #968.

**Comment by edhartnett** [comment]
These changes have made it into master, so I will close this ticket.
